pilz_control/include/pilz_control/traj_mode_manager.h

```diff
@@ -62,65 +62,57 @@ class TrajProcessingModeStateMachine
 };
 
 /**
- * @brief Listener to wait for a specified mode to be reached.
+ * @brief Listener to wait for the hold mode to be reached.
  */
-class TrajProcessingModeListener
+class HoldModeListener
 {
 public:
-  //! @param mode Mode to wait for.
-  TrajProcessingModeListener(const TrajProcessingMode& mode);
-  void waitForMode();
-  //! @returns true if the given mode corresponds to the target mode, otherwise false.
-  bool isTargetMode(const TrajProcessingMode& mode) const;
-  //! @brief Notify the listener that the target mode is reached.
+  void waitForHold();
+  //! @brief Notify the listener that the hold mode is reached.
   void triggerListener();
 
 private:
   std::mutex mutex_;
   std::condition_variable cond_variable_;
   bool cond_fulfilled_ {false};
-  //! @brief Mode to wait for.
-  const TrajProcessingMode mode_;
 };
 
 /**
  * @brief Encapsulates a state machine managing the current Trajectory-Processing-Mode.
+ * 
+ * All events are mutually exclusive. Initial mode is stopping in accordance to the controller.
  */
 class TrajProcessingModeManager
 {
 public:
-  //! @returns true only if a successful state switch to stopping was performed, otherwise false.
-  bool stoppingEvent();
+  //! @return True only if a successful state switch to stopping was performed, otherwise false.
+  bool stopEvent(HoldModeListener* const listener = nullptr);
   //! @brief Switch to hold.
   void stopMotionFinishedEvent();
-  //! @returns true if in state unhold or a successful switch to state unhold was performed, otherwise false.
-  bool unholdEvent();
+  //! @return True if in state unhold or a successful switch to state unhold was performed, otherwise false.
+  bool startEvent();
 
 public:
   //! @brief Check if in state stopping or hold.
   bool isHolding();
-  bool isUnhold();
-  void registerListener(TrajProcessingModeListener* const listener);
-
   TrajProcessingMode getCurrentMode();
 
 private:
-  //! @brief Perform full transition if possible.
-  bool switchTo(const TrajProcessingMode& mode, const bool success_at_transition_only=true);
-  bool setMode(const TrajProcessingMode& requested_mode, const bool& success_at_transition_only);
+  /**
+   * @brief Perform transition if possible.
+   * @return True if transition was performed, otherwise false.
+   */
+  bool switchTo(const TrajProcessingMode& mode);
+  void registerHoldListener(HoldModeListener* const listener);
```

Comment:
The name is redundant because the argument type already indicates that only `HoldModeListener` can be registered.

Reply:
Done.